edit/lint.js

```diff
@@ -0,0 +1,413 @@
+    }
+  });
+  return invalid;
+}
+
+function setupLinterSettingsEvents(popup) {
+  $('.save', popup).addEventListener('click', event => {
+    event.preventDefault();
+    const linter = checkLinter(event.target.dataset.linter);
+    const json = tryJSONparse(popup.codebox.getValue());
+    if (json && json.rules) {
+      const invalid = checkRules(linter, json.rules);
+      if (invalid.length) {
+        return showLinterErrorMessage(
+          linter,
+          t('setLinterInvalidRuleError') + `<ul><li>${invalid.join('</li><li>')}</li></ul>`
+        );
+      }
+      if (linter === 'stylelint') {
+        setStylelintRules(json.rules);
+      } else {
+        setCSSLintRules(json.rules);
+      }
+      updateLinter(linter);
+      showSavedMessage();
+    } else {
+      showLinterErrorMessage(linter, t('setLinterError'));
+    }
+  });
+  $('.reset', popup).addEventListener('click', event => {
+    event.preventDefault();
+    const linter = checkLinter(event.target.dataset.linter);
+    let rules;
+    if (linter === 'stylelint') {
+      setStylelintRules();
+      rules = {rules: stylelintDefaultConfig.rules};
+    } else {
+      setCSSLintRules();
+      rules = {rules: csslintDefaultRuleset};
+    }
+    popup.codebox.setValue(JSON.stringify(rules, null, 2));
+    updateLinter(linter);
+  });
+}
+
+function openStylelintSettings() {
+  const linter = prefs.get('editor.linter');
+  BG.chromeSync.getValue(
+    linter === 'stylelint'
+      ? 'editorStylelintRules'
+      : 'editorCSSLintRules'
+  ).then(rules => {
+    if (!rules || rules.length === 0) {
+      rules = linter === 'stylelint'
+        ? setStylelintRules(rules)
+        : setCSSLintRules(rules);
+    }
+    const rulesString = JSON.stringify({rules: rules}, null, 2);
+    setupLinterPopup(rulesString);
+  });
+}
+
+function setupLinterPopup(rules) {
+  const linter = prefs.get('editor.linter');
+  const linterTitle = linter === 'stylelint' ? 'Stylelint' : 'CSSLint';
+  function makeButton(className, text) {
+    return $element({tag: 'button', className, type: 'button', textContent: t(text), dataset: {linter}});
+  }
+  function makeLink(url, textContent) {
+    return $element({tag: 'a', target: '_blank', href: url, textContent});
+  }
+  function setJSONMode(cm) {
+    cm.setOption('mode', 'application/json');
+    cm.setOption('lint', 'json');
+  }
+  const popup = showCodeMirrorPopup(t('setLinterRulesTitle', linterTitle), $element({
+    appendChild: [
+      $element({
+        tag: 'p',
+        appendChild: [
+          t('setLinterLink') + ' ',
+          makeLink(
+            linter === 'stylelint'
+              ? 'https://stylelint.io/demo/'
+              : 'https://github.com/CSSLint/csslint/wiki/Rules-by-ID',
+            linterTitle
+          ),
+          linter === 'csslint' ? ' ' + t('showCSSLintSettings') : ''
+        ]
+      }),
+      makeButton('save', 'styleSaveLabel'),
+      makeButton('reset', 'genericResetLabel'),
+      $element({
```

Comment:
The config dialog has `Save` and `Reset` buttons so I think it should also have a `Cancel` button (i18n message is `confirmCancel`) in order to present all available actions in one place.

Reply:
Do you think the reset button should have a confirmation popup? Or maybe move the button to the right side to minimize chances of accidental clicking?

Reply:
Hmm, you can add a tooltip "To undo accidental reset, simply press Ctrl-Z (or Cmd-Z) in the text box".